internal/jennies/java/templates/converters/converter.tmpl

```diff
@@ -0,0 +1,181 @@
+package {{ .Converter.Package | formatPackageName }};
+
+{{ $converter := include "converter" . }}
+{{- .Imports }}
+{{ $converter }}
+
+{{- define "guard" }}
+    {{- $leftOperand := print (.Path | formatGuardPath) -}}
+    {{- $type := .Path.Last.Type }}
+    {{- if and (eq .Op "!=") (eq .Value nil) -}}
+        {{ $leftOperand }} != null
+    {{- else if and $type.IsScalar (eq $type.AsScalar.ScalarKind "bool") -}}
+        {{- if not $type.Nullable }}
+        {{- $leftOperand }} != null && !{{- $leftOperand }}
+        {{- else }}!{{- $leftOperand }}
+        {{- end }}
+    {{- else if and $type.IsScalar (eq $type.AsScalar.ScalarKind "string") -}}
+        {{- if .Value }}!{{- $leftOperand }}.equals("{{ cleanString .Value }}")
+        {{- else if not $type.Nullable }}
+            {{- $reflect := importStdPkg "java.util" "Objects" -}}
+            {{- $leftOperand }} != null && !Objects.equals({{- $leftOperand }}, "")
+        {{- else }}
+            {{- $reflect := importStdPkg "java.util" "Objects" -}}
+            !Objects.equals({{- $leftOperand }}, "")
+        {{- end }}
+    {{- else if $type.IsRef }}
+        {{- formatRefType $type .Value }}
+    {{- else if $type.IsArray }}
+        {{- if not $type.Nullable }}
+        {{- $leftOperand }} != null && !{{- $leftOperand }}.isEmpty()
+        {{- else }}!{{- $leftOperand }}.isEmpty()
+        {{- end }}
+    {{- else -}}
+        {{- $operator := .Op -}} 
+        {{- if eq .Op "minLength" -}}
+            !{{- $leftOperand }}.isEmpty()
+        {{- else }}
+            {{- if eq .Op "maxLength" -}}
+                {{- $leftOperand = print $leftOperand ".size()" -}}
+                {{- $operator = "<=" -}}
+            {{- end }}
+    
+            {{- $leftOperand }} {{ $operator }} {{ .Value | formatScalar -}}
+        {{- end }}
+    {{- end -}}
+{{- end }}
+
+{{- define "value_formatter" -}}
+    {{- if .IsMapEntry }}
+    {{- .Path | formatGuardPath }}
+    {{- else if .Type.IsScalar }}
+        {{- if and (eq .Type.AsScalar.ScalarKind "string") -}}
+        {{- $reflect := importStdPkg "cog" "Runtime" -}}
+        Runtime.escapeString({{- .Path | formatGuardPath }})
+        {{- else }}
+        {{- .Path | formatGuardPath }} + "{{ formatIntegerLetter .Type }}"
+        {{- end }}
+    {{- else if resolvesToEnum .Type -}}
+        {{- $reflect := importStdPkg "cog" "Runtime" -}}
+        {{- $reflect := importStdPkg .Type.AsRef.ReferredPkg .Type.AsRef.ReferredType -}}
+        Runtime.formatEnum({{ .Type.AsRef.ReferredType }}.class, {{- .Path | formatGuardPath }})
```

Comment:
For enums we need to know the class because if the value is null, we cannot know the class. In Java we cannot use the string or the integer directly. We need to set `MyEnumClass.DEFAULT_VALUE`.